lesson_7/trif_mihaela/lesson_7_trifmihaela.py

```diff
@@ -0,0 +1,55 @@
+# Create a decorator called uppercase that will uppercase the result
+ 
+def decorator_uppercase(my_function):
+
+    def inner_uppercase(*arg):
+        return my_function(*arg).upper()
+    return inner_uppercase
+
+@decorator_uppercase
+def greet(name):
+    return "Greetings {}!".format(name)
+
+print(greet('World'))
+
+# 2. # Create a decorator called safe_divide that will output a message if the division cannot be performed, othervise it will return the result.
+def decorator_safe_divide(my_function):
+
+    def inner_safe_divide(*args):
+        if my_function(*args) is None:
+            print('The divizion cannot be preformed')
+        return my_function(*args)
+    return inner_safe_divide
+
+@decorator_safe_divide
+def divide(first_number, second_number):
+     return first_number / second_number
+
+print(divide(12,15))
+# 3.Create a decorator called register that will update a list called print_registry with all the decorated functions names.
+def register(my_function):
+    
+    def inner_registry(*args):
+        my_function_name = my_function.__name__
```

Comment:
You can make the append to `print_registry` inside the decorator. In this way, you can return `my_function` and keep the functionality of the decorated function and just register the function in `print_registry`. 

Currently you will lose the functionality of your function and just return it's name 

```
print_registry = []
def register(my_function):

    def inner_registry(*args):
        my_function_name = my_function.__name__
        print_registry.append(my_function_name)
        return my_function
    return inner_registry
```